storage/innobase/dict/dict0dd.cc

```diff
@@ -813,7 +813,23 @@ dict_table_t *dd_table_create_on_dd_obj(const dd::Table *dd_table,
       charset_no = my_charset_bin.number;
     else if (dd_col->type() == dd::enum_column_types::JSON)
       charset_no = my_charset_utf8mb4_bin.number;
-    else if (dtype_is_string_type(mtype)) {
+    else if (dtype_is_real_temporal_type(dd_col->type())) {
+      /*
+        MySQL always stores real temporal fields in my_charset_numeric charset.
```

Comment:
I think this should be mentioned in is_real_temporal_type function comment.

Reply:
IMO this comment may not be appropriate for function comment since the function's responsibility is only to check if the data type is temporal or not. So I prefer to have it here where we take decision.